Oracle_CQRS/src/main/java/org/example/oracle/cqrs/command/commands/CreateAccountCommand.java

```diff
@@ -0,0 +1,17 @@
+package org.example.oracle.cqrs.command.commands;
+
+import lombok.*;
+
+@Getter @NoArgsConstructor
+public class CreateAccountCommand extends BaseCommand<String> {
+    private double initialBalance;
+    private String currency;
+    private String accountId ;
+
+    public CreateAccountCommand(String id, double initialBalance, String currency, String accountId) {
+        super(id);
+        this.initialBalance = initialBalance;
+        this.currency = currency;
+        this.accountId = accountId;
+    }
```

Comment:
public BaseEvent createEvent() {

                System.out.println("Handling create: " + command);
                if (createAccountCommand.getInitialBalance() < 0)
                    throw new IllegalArgumentException("Initial balance is negative");

                event = new AccountCreatedEvent(UUID.randomUUID().toString(), createAccountCommand.getInitialBalance(), createAccountCommand.getCurrency(), AccountStatus.CREATED, createAccountCommand.getAccountId());
}